src/Data/Codec/Argonaut/Sum.purs

```diff
@@ -150,6 +154,44 @@ finalizeError name err =
 
 data Err = UnmatchedCase | JErr JsonDecodeError
 
+sum' ∷ ∀ r' r rep a. GTagsMap r' rep ⇒ GCases r rep ⇒ Generic a rep ⇒ String → Record r' → Record r → JsonCodec a
+sum' = sumWith' defaultEncoding
+
+sumWith' ∷ ∀ r' r rep a. GTagsMap r' rep ⇒ GCases r rep ⇒ Generic a rep ⇒ Encoding → String → Record r' → Record r → JsonCodec a
+sumWith' encoding name tags r =
+  sumWith encoding' name r
+  where
+  mapObj = gTagsMap (Proxy @rep) tags
+  useTagsMap s = fromMaybe s (Obj.lookup s mapObj)
+  encoding' = case encoding of
+    EncodeNested enc →
+      EncodeNested $ enc { mapTag = useTagsMap >>> enc.mapTag }
+
+    EncodeTagged enc →
+      EncodeTagged $ enc { mapTag = useTagsMap >>> enc.mapTag }
+
+--------------------------------------------------------------------------------
+
+class GTagsMap ∷ ∀ k. Row Type → k → Constraint
+class GTagsMap r rep where
+  gTagsMap ∷ Proxy rep → Record r → Object String
```

Comment:
How about defining the class without method if the only thing it does is unsafe coerce?
This operation could be done once in `sumWith` where its called.

Another thought: You could avoid the `unsafeCoerce` if you add the `Homogeneous` constraint and then use: https://pursuit.purescript.org/packages/purescript-foreign-object/4.1.0/docs/Foreign.Object#v:fromHomogeneous

Reply:
Potentially we could construct this `Object` safely, also we could use `Map`, Object just for speed. `Object.fromHomogeneous` is unsafeCoerce anyway, though if we add `Homogeneous r String` constraint we can use `fromHomogeneous`. I also thought about how this would work in the new `codec-json` lib, should we convert this mapping record to safely constructed `Map String String` or unsafely coerced `JObject`?

Reply:
> I also thought about how this would work in the new `codec-json` lib, should we convert this mapping record to safely constructed `Map String String` or unsafely coerced `JObject` (for speed)?

(I know this isn't necessarily the place to discuss it just here, but `unsafeCoerce` is not allowed in `codec-json` - we can implement something in the FFI to do the conversion and just have it return the input, so essentially `unsafeCoerce`, but using `unsafeCoerce` directly is a no-go. The purpose of `JObject`, `JArray`, etc. is to uncouple the representation of JSON from the representation the language uses for map-like and array-like things, so that in theory it's usable by other backends too where the shared representation may not be viable).

Reply:
Well, really this tags' map is constructed once when the codec is declared, so I believe we can just use `Map String String` (if `Object String` is not available) here, it shouldn't have much impact with `Map.lookup` when converting tags.